.github/workflows/flaky.yaml

```diff
@@ -0,0 +1,46 @@
+# Run all tests, including flaky test.
+#
+# The default CI workflow ignores flaky tests.  This workflow will run
+# all tests, including ignored ones.
+#
+# To use this workflow you can either:
+#
+# - Label a PR with "flaky-test", the normal CI workflow will not run
+#   any jobs but the jobs here will be run.  Note that to merge the PR
+#   you'll need to remove the label eventually because the normal CI
+#   jobs are required by branch protection.
+#
+# - Manually trigger the workflow, you may choose a branch for this to
+#   run on.
+#
+# Additionally this jobs runs once a day on a schedule.
+#
+# Currently doctests are not run by this workflow.
+
+name: Flaky CI
+
+on:
+  pull_request:
+    types: [ 'labeled', 'opened', 'synchronize', 'reopened' ]
+  schedule:
+    # 06:30 UTC every day
+    - cron: '30 6 * * *'
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to run on, defaults to main'
+        required: true
+        default: 'main'
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  tests:
+    if: "contains(github.event.pull_request.labels.*.name, 'flaky-test') || contains(github.event, 'workflow_dispatch')"
```

Comment:
This can be done `github.event_name == 'workflow_dispatch'` for a bit more precision.
Also doesn't cover the `schedule` event and will basically skip it altogether on every trigger.